extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -588,88 +186,80 @@ export async function newDeployment(
-      discoveredEntryPoints = entryPointsResponse.data;
-
-      // build up a list of open files, relative to the opened workspace folder
-      const openFileList: string[] = vscodeOpenFiles();
-
-      // loop through and now separate possible entrypoints into open or not
-      discoveredEntryPoints.forEach((entrypointFile) => {
-        if (
-          openFileList.find(
-            (f) => f.toLowerCase() === entrypointFile.toLowerCase(),
-          )
-        ) {
-          entrypointFilesOpened.push(entrypointFile);
-        } else {
-          entrypointFilesUnopened.push(entrypointFile);
-        }
+    if (entryPointFile) {
+      // we were passed in a specific entrypoint file.
+      // while we don't need it, we'll still provide the results
+      // in the same way.
+      const entryPointPath = path.join(projectDir, entryPointFile);
+      entryPointListItems.push({
+        iconPath: new ThemeIcon("file"),
+        label: entryPointPath,
       });
+      return resolve();
+    }
 
-      // build the entrypointList
-      if (entrypointFilesOpened.length) {
-        entryPointListItems.push({
-          label: "Open Files",
-          kind: QuickPickItemKind.Separator,
-        });
-        entrypointFilesOpened.forEach((entryPointFile) => {
-          entryPointListItems.push({
-            iconPath: new ThemeIcon("file"),
-            label: entryPointFile,
-          });
-        });
+    // build up a list of open files, relative to the opened workspace folder
+    const rawOpenFileList: string[] = vscodeOpenFiles();
+    const filteredOpenFileList: string[] = [];
+    rawOpenFileList.forEach((openFilePath) => {
+      const parsedPath = path.parse(openFilePath);
+      if (ENTRYPOINT_FILE_EXTENSIONS.includes(parsedPath.ext.toLowerCase())) {
+        filteredOpenFileList.push(openFilePath);
       }
-      if (entrypointFilesUnopened.length) {
-        entryPointListItems.push({
-          label: "Discovered Entrypoints",
-          kind: QuickPickItemKind.Separator,
-        });
-        entrypointFilesUnopened.forEach((entryPointFile) => {
-          entryPointListItems.push({
-            iconPath: new ThemeIcon("file"),
-            label: entryPointFile,
+    });
+    filteredOpenFileList.sort();
+
+    // build the entrypointList
+    if (filteredOpenFileList.length) {
+      entryPointListItems.push({
+        label: "Open Files",
+        kind: QuickPickItemKind.Separator,
+      });
+      const python = await getPythonInterpreterPath();
+      for (const openFile of filteredOpenFileList) {
+        // inspect each file type
+        try {
+          const relEntryPointDir = path.dirname(openFile);
+          const relEntryPointFile = path.basename(openFile);
+
+          const inspectResponse = await api.configurations.inspect(
+            relEntryPointDir,
+            python,
+            {
+              entrypoint: relEntryPointFile,
+            },
+          );
```

Comment:
I thought we only did inspection once the file was selected. With the open file list now I see the same file for each content type is available to it, which I don't think is what we want:

![CleanShot 2024-09-25 at 15 17 39@2x](https://github.com/user-attachments/assets/b4f131eb-0875-4cda-9d6e-72fef32ebff5)

Reply:
Just double checked, if I select one of the `static.Rmd` files above I still get the content type question

Reply:
Per our discussion, this has been refactored to return to the behavior you described.